### ISSUE TITLE
Add example `with-backend`

### ISSUE THREAD

**Issue by ifiokjr** [remirror/remirror#457]
### Description

Show how the editor can be used with a backend for storing data and persisting images and working with collaboration.

**Comment by anotheri** [comment]
@ifiokjr just to confirm, is `remirror/extension/collaboration` ready to be used? any chance I can get examples or docs on how to implement that right way?
thanks in advance.

**Comment by anotheri** [comment]
or should I use `yjs` extension instead? or both?

**Comment by ifiokjr** [comment]
@anotheri yjs is your best bet, although there's no documentation at the moment on how to use it.

**Comment by anotheri** [comment]
Thanks, I'll try to figure that out.

**Comment by anotheri** [comment]
@ifiokjr another question: is it possible to make it work with Socket.Io? or I have to use WebSockets?

**Comment by ifiokjr** [comment]
I haven't really looked into the mechanics of `yjs` but I'm sure it's possible.

It's something I'm planning to look at in more detail eventually.

**Comment by anotheri** [comment]
They have `y-websockets`, which doesn't work with the socket.io server as far as I can see. I've also noticed that their server implementation for `y-websockets` built on top of WebSockets. So was wondering if there is a kind of ready-to-go config to make it work with socket.io instead. Since it uses a different API, I can't use the same provider implementation.
And I'm trying to switch my server to WebSockets now. It could be easier imo.

**Comment by ditschedev** [comment]
@anotheri socket.io is an abstraction of WebSockets using a custom protocol. So only socket.io clients may work with a socket.io server and vice versa. You'd need another WebSocket server (e.g [ws](https://github.com/websockets/ws) or [uWebSockets](https://github.com/uNetworking/uWebSockets.js)

**Comment by anotheri** [comment]
@ditschedev yeah, already figured that out. And switched to WS. Although it's possible to write your own provider which uses socket.io as alternative.
Anyway thanks for comment.
